### ISSUE TITLE
Applying .AsCached(new SqlDependencyInvalidator()) to EF 6.1.1 context throws exception

### ISSUE THREAD

**Issue by Chudokranik** [osjoberg/LinqCache#1]
Hi Oskar!

I am using EF 6.1.1 and tried to perform .AsCached operation with SqlDependencyInvalidator provided in the following manner:

```
            var exemptionIncomeLimits =
                context.ExemptionIncomeLimits
                .AsCached(new SqlDependencyInvalidator())
                .ToList();
```

where context is a class inherited from DbContext (usual EF context)

As a result I got exception from OnInit right here:

```
        var contextFieldInfo = queryType.GetField("context", BindingFlags.NonPublic | BindingFlags.Instance);
        if (contextFieldInfo == null)
        {
            throw LinqCacheException.ContextIsNotSupported;
        }
```

Could you please explain what I am doing wrong? Btw, using AsCached without invalidator works just fine.

Thank you!

**Comment by osjoberg** [comment]
Hello Chudokranik!

Thank you for reporting this issue. The SqlDependencyInvalidator was only implemented for Linq-to-SQL, however the current implementation is broken and more code is required to make it work for Entity Framework. Also there are reports that SqlDependency wich SqlDependencyInvalidator uses in combination with Entity Framework is not so useful because Entity Framework queries oftenly becomes SQL subqueries which SQL-Server does not support dependency tracking for. However I will investigate and see if Entity Framework support is feasible.

Best regards,
Oskar

**Comment by osjoberg** [comment]
I just want to notify you that EntityFramework support for SqlDependencyInvalidator is implemented as of version 0.1.5. Enjoy!
